packages/fx-core/src/common/tools.ts

```diff
@@ -455,11 +455,12 @@ export function canAddSso(
       : false;
   }
 
-  const solutionSettings = projectSettings.solutionSettings as AzureSolutionSettings;
-  if (
-    !projectSettings.solutionSettings ||
-    !(solutionSettings.hostType === HostTypeOptionAzure.id)
-  ) {
+  const solutionSettings = projectSettings.solutionSettings;
+  if (!solutionSettings) {
```

Comment:
How about using `isExistingTab(projectSettings)` to check if it's exisiting tab, and then check if SSO already added with `solutionSettings.capabilities.includes(TabSsoItem.id)`?

Reply:
OK, so it would be like this?
```
if (isExistingTabApp(projectSettings) && !solutionSettings.capabilities.includes(TabSsoItem.id)) {
  return ok(Void);
}
```

Reply:
yeah, it looks good to me 😊